report.c

```diff
@@ -399,49 +400,15 @@ void report_voltage()
 void calculate_motor_voltage(){
   uint8_t i;
 
-  // Set the ADC Reference
-  ADMUX = (1<<REFS0);
-  // Enable ADC, start conversion, prescaler of 128x
-  ADCSRA = (1<<ADEN)|(1<<ADPS2)|(1<<ADPS1)|(1<<ADPS0);
-
   for (i=0;i<(VOLTAGE_SENSOR_COUNT-1); i++){
-    // If MUX5 bit was set high, it must be cleared
-    // in order for the other analogs to read properly
-    if (MUX5_BIT_VALUE)
-      ADCSRB &= ~(MUX5_BIT_VALUE<<MUX5_BIT_POS);
-
-    // set next motor target for ADC
-    ADMUX = (1<<REFS0) + i;
-
-    // Enable capture of ADC
-    ADCSRA |= (1<<ADSC);
-    // Wait for the result
-    while(ADCSRA & (1<<ADSC));
-    // Store result
-    analog_voltage_readings[i] = ADC;
+    analog_voltage_readings[i] = adc_read_channel(i); // Assumes the motors are on ADC channels 0-3 and 
+                                                      // in the same order in analogvoltage_readings
   }
 }
 
 // Calculates force sensor value.
 void calculate_force_voltage(){
-  // Set the ADC Reference
-  ADMUX = (1<<REFS0);
-
-  // Enable ADC, start conversion, prescaler of 128x
-  ADCSRA = (1<<ADEN)|(1<<ADPS2)|(1<<ADPS1)|(1<<ADPS0);
-
-  // Set read bit to be the special feedback bit (ADC14)
-  // This is bit select 6, plus
-  ADMUX = ((1<<REFS0) + (FVOLT_ADC));
-  ADCSRB = (MUX5_BIT_VALUE<<MUX5_BIT_POS);
-  // Enable capture of ADC
-  ADCSRA |= (1<<ADSC);
-  // Wait for the result
-  while(ADCSRA & (1<<ADSC));
-  // Store the result
-  analog_voltage_readings[FORCE_VALUE_INDEX] = ADC;
-  // Disable ADC
-  ADCSRA = (0<<ADEN);
+  analog_voltage_readings[FORCE_VALUE_INDEX] = adc_read_channel(F_VOLT_ADC); 
```

Comment:
We should do a study on this for repeatability to ensure we don't need any sort of software filtering of the ADC reads.